go/internal/controller/translator/adk_api_translator.go

```diff
@@ -511,6 +558,15 @@ func (a *adkApiTranslator) translateInlineAgent(ctx context.Context, agent *v1al
 					Url:         url,
 					Description: toolAgent.Spec.Description,
 				})
+			case v1alpha2.AgentType_Remote:
+				/* TODO: Add support for remote agents.
```

Comment:
The translator is a pure function, if we are going to make a network call we should do it in the reconciler

Reply:
Makes sense. Just need to figure out how the polling of agent card ties into this a bit better. I was/am planning on handling polling logic in a follow-up to avoid blowing up this PR with diffs.

Self-thought/note:
I _think_ we do something along these lines for MCP tools, where there's a watcher that updates agents when mcp tools update. So something like that would be done for remote agent-as-a-tool, where updates would occur when they get re-polled.